xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp

```diff
@@ -746,9 +746,13 @@ bool CAESinkALSA::InitializeHW(const ALSAConfig &inconfig, ALSAConfig &outconfig
       int bits    = snd_pcm_hw_params_get_sbits(hw_params);
       if (bits != fmtBits)
       {
-        /* if we opened in 32bit and only have 24bits, signal it accordingly */
+        /*
+         if we opened in 32bit and only have 24bits, signal it accordingly
+         S24NE4MSB only works with S32 alsa format, use S24NE4 for S24 alsa format
+         skip format for all other sbits values as there is no matching AE format
+        */
         if (fmtBits == 32 && bits == 24)
-          i = AE_FMT_S24NE4MSB;
+          i = (fmt == SND_PCM_FORMAT_S32) ? AE_FMT_S24NE4MSB : AE_FMT_S24NE4;
```

Comment:
the entire block has become rather ugly. better code what you wrote in the comments:
if (SND_PCM_FORMAT_S32 && 24)
else if (SND_PCM_FORMAT_S24 && 24)
else